src/v8/inspector/server.rs

```diff
@@ -0,0 +1,852 @@
+
+        Self(value)
+    }
+}
+
+struct InspectorCallbacks {
+    on_response: Box<OnResponseCallback>,
+    on_wait_frontend_message_on_pause: Box<OnWaitFrontendMessageOnPauseCallback>,
+}
+
+/// The means of connection to the V8 debugger.
+#[derive(Debug, Clone)]
+pub struct DebuggerSessionConnectionHints {
+    address: std::net::SocketAddr,
+    vscode_configuration: String,
+    chromium_link: String,
+}
+
+impl DebuggerSessionConnectionHints {
+    /// Returns a [std::net::SocketAddr] address on which the server is
+    /// going to be listening.
+    pub fn get_address(&self) -> std::net::SocketAddr {
+        self.address
+    }
+
+    /// Returns a hint on how to connect to the remote debugger server
+    /// for remote debugging using Visual Studio Code.
+    pub fn get_visual_studio_code_configuration(&self) -> &str {
+        &self.vscode_configuration
+    }
+
+    /// Returns a hint on how to connect to the remote debugger server
+    /// for remote debugging using Chromium-based browsers.
+    pub fn get_chromium_link(&self) -> &str {
+        &self.chromium_link
+    }
+}
+
+impl std::fmt::Display for DebuggerSessionConnectionHints {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let address = &self.address;
+        let vscode_configuration = &self.vscode_configuration;
+        let chromium_link = &self.chromium_link;
+        f.write_fmt(format_args!("The V8 remote debugging server is waiting for a connection via WebSocket on: {address}.\nHint: you may browse this link: {chromium_link} or use this launch configuration for Visual Studio Code (<https://code.visualstudio.com/>):{vscode_configuration}"))
+    }
+}
+
+impl<T: Into<std::net::SocketAddr>> From<T> for DebuggerSessionConnectionHints {
+    fn from(address: T) -> Self {
+        let address = address.into();
+        let vscode_configuration = format!(
+            r#"
+        {{
+            "version": "0.2.0",
+            "configurations": [
+                {{
+                    "name": "Attach to RedisGears V8 through WebSocket at {address}",
+                    "type": "node",
+                    "request": "attach",
+                    "cwd": "${{workspaceFolder}}",
+                    "websocketAddress": "ws://{address}",
+                }}
+            ]
+        }}
+        "#
+        );
+        let chromium_link = format!(
+            "devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws={address}"
+        );
+        DebuggerSessionConnectionHints {
+            address,
+            vscode_configuration,
+            chromium_link,
+        }
+    }
+}
+
+/// A single debugger session.
+#[derive(Debug)]
+pub struct DebuggerSession {
+    web_socket: Rc<Mutex<WebSocketServer>>,
+    inspector: Inspector,
+    connection_hints: DebuggerSessionConnectionHints,
+}
+
+impl DebuggerSession {
+    fn create_inspector_callbacks(web_socket: Rc<Mutex<WebSocketServer>>) -> InspectorCallbacks {
```

Comment:
Why do we need an Rc of Mutex here? If its for thread safety shouldn't it be Arc? But I believe maybe we do not need them at all if we count of the thread safely of the isolate no? Both `on_response` and `on_wait_frontend_message_on_pause` are called when the isolate is entered.

Reply:
What about this one?